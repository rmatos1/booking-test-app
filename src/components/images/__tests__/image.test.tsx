import { describe, test } from '@jest/globals';
import { render } from '@testing-library/react';

import { IImage, Image } from '..';

const defaultProps: IImage = {
  imgSrc: './test.jpg',
};

const setup = (componentProps?: IImage): JSX.Element => {
  const baseProps = componentProps || defaultProps;

  return <Image {...baseProps} />;
};

describe('<Image />', () => {
 
  test('should render correctly an image with the defined src attribute', () => {
    const wrapper = render(setup());

    const img = wrapper.getByTestId('image');

    expect(img.getAttribute('src')).toBe(defaultProps.imgSrc);
   
    expect(wrapper).toMatchSnapshot();
  });

  test('should render correctly an image with the defined alt attributes', () => {

    const ALT = "alt-test";

    const wrapper = render(setup({ ...defaultProps, alt: ALT }));

    const img = wrapper.getByTestId('image');

    expect(img.getAttribute('alt')).toBe(ALT);

    expect(wrapper).toMatchSnapshot();
  });
});
