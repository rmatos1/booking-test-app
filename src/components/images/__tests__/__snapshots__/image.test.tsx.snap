// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> should render correctly an image with the defined alt attributes 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        alt="alt-test"
        class="sc-bdfCDU fXjOdt"
        data-testid="image"
        src="./test.jpg"
      />
    </div>
  </body>,
  "container": <div>
    <img
      alt="alt-test"
      class="sc-bdfCDU fXjOdt"
      data-testid="image"
      src="./test.jpg"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Image /> should render correctly an image with the defined src attribute 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        alt=""
        class="sc-bdfCDU fXjOdt"
        data-testid="image"
        src="./test.jpg"
      />
    </div>
  </body>,
  "container": <div>
    <img
      alt=""
      class="sc-bdfCDU fXjOdt"
      data-testid="image"
      src="./test.jpg"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
